### PR TITLE
Get table completion percentage

### LINKED ISSUE #36: Add percentage of filled values for columns
It's useful to know if the column is filled at 90% or 5%
This could be integrated in the current view to explore a table.

### PR DESCRIPTION
**Pull request by elsiehoffet-94**
First attempt to compute column completion percentage on a table of a database with a simple sql query `select count(column)/count(*) from table`. 

Closes #36

### CODE REVIEW

File: get_db_stats.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from os import getenv
+
+from pagai.services.database_explorer import POSTGRES, ORACLE, DatabaseExplorer
+from pagai.errors import OperationOutcome
+from sqlalchemy.exc import OperationalError
+import json
+
+def get_col_completion(owner, table_name, sorted, db_model):
```
Comment (Jasopaum) [inline get_db_stats.py:8]: Where will this be called? Do you think we should make it a route (as for get_db_schema)?

Comment (Jasopaum) [inline get_db_stats.py:8]: When do you think `sorted` will be useful?

File: get_db_stats.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    Returns the percentage of completion for all columns in the given table
+    """
+
+    # switch on the possible db models
+    # if the db model is not supported, an error is raised.
+    db_drivers = {"POSTGRES": POSTGRES, "ORACLE": ORACLE}
+
+    credentials = {
+        'model': 'public',
```
Comment (Jasopaum) [inline get_db_stats.py:18]: `public`?

File: pagai/services/database_explorer.py [hunk 3]
```diff
@@ -134,3 +134,20 @@ def get_db_schema(self, owner: str, driver=POSTGRES):
...
+    def get_column_completion(self, db_schema: defaultdict(list), table: str, sort: bool):
+        """
+        Returns a list of (column_name, percentage of completion) for all columns in a given table
+        """
+        query_tmp = ""
+        column_list = sorted(db_schema[table]) if sort else db_schema[table]
+        for column in column_list[:-1] :
+            query_tmp += f"round(count({column}) / count(*) * 100, 0), "
+        query_tmp += f"round(count({column_list[-1]}) / count(*) * 100, 0) "
+        sql_query = f"select {query_tmp} from {table}"
```
Comment (Jasopaum) [inline pagai/services/database_explorer.py:148]: Do you think we could do that with sqlAlchemy?

**Comment by simonvadee** [comment]
stale

**Pull request closed without merging**